add freertos-addons and option to set path to portable directory

johnathancn opened the pull request.

johnathancn commented:
#35 #34

schteve commented on freertos-cargo-build/src/lib.rs:
```diff
@@ -61,6 +62,7 @@ impl Builder {
         let freertos_path = env::var(ENV_KEY_FREERTOS_SRC).unwrap_or_default();
         let freertos_config_path = env::var(ENV_KEY_FREERTOS_CONFIG).unwrap_or_default();
         let freertos_shim = env::var(ENV_KEY_FREERTOS_SHIM).unwrap_or_default();
+        let freertos_port_base = format!("{}{}", freertos_path, "portable");
```
Would prefer to use something like `Path::join` here to ensure the separator is correct in all cases.

schteve commented:
Thanks for this, could you please point the submodule to the latest released tag from the other repo? I would prefer to follow releases for simplicity.

Also, based on our conversation the fix for #35 is probably more involved than just adding a define. Feel free to remove those changes for now if you want to merge sooner.

johnathancn commented:
> Thanks for this, could you please point the submodule to the latest released tag from the other repo? I would prefer to follow releases for simplicity.
> 
> Also, based on our conversation the fix for #35 is probably more involved than just adding a define. Feel free to remove those changes for now if you want to merge sooner.

Updated the tag to v1.6.0 and removed the #define patch

schteve commented:
Looks good to me. Thanks again!
